src/snowcli/cli/nativeapp/manager.py

```diff
@@ -309,18 +326,21 @@ def _create_dev_app(self, diff: DiffResult) -> None:
                         self.app_name, self.app_role, actual_owner
                     )
 
-                if diff.has_changes():
-                    # the app needs to be upgraded
-                    log.info(f"Upgrading existing application {self.app_name}.")
+                try:
+                    if diff.has_changes():
+                        # the app needs to be upgraded
+                        log.info(f"Upgrading existing application {self.app_name}.")
+                        self._execute_query(
+                            f"alter application {self.app_name} upgrade using @{self.stage_fqn}"
+                        )
+
+                    # ensure debug_mode is up-to-date
                     self._execute_query(
-                        f"alter application {self.app_name} upgrade using @{self.stage_fqn}"
+                        f"alter application {self.app_name} set debug_mode = {self.debug_mode}"
                     )
-
-                # ensure debug_mode is up-to-date
-                self._execute_query(
-                    f"alter application {self.app_name} set debug_mode = {self.debug_mode}"
-                )
-                return
+                    return
+                except ProgrammingError as err:
+                    self._raise_missing_warehouse_err(err)
```

Comment:
At this moment you don't know if it is `missing warehouse error`, I would suggest a more general name such as `_raise_err`